voltdbclient/distributor.go

```diff
@@ -314,19 +311,20 @@ func (c *Conn) loop(disconnected []*nodeConn, hostIDToConnection *map[int]*nodeC
 }
 
 func (c *Conn) submit(pi *procedureInvocation) (int, error) {
-	var nc *nodeConn
-	var backpressure = true
-	var err error
-	if c.useClientAffinity && c.hnator != nil && c.partitionReplicas != nil && c.procedureInfos != nil {
-		nc, backpressure, err =
-			c.getConnByCA(c.connected, c.hnator, &c.partitionMasters, c.partitionReplicas, c.procedureInfos, pi)
-	}
-	if err != nil && !backpressure && nc != nil {
-		// nc.submit(pi)
-	} else {
-		// c.allNcsPiCh <- pi
-	}
-	return c.subscribedConnection.submit(pi)
+	nc := c.availableConn()
+	// var nc *nodeConn
+	// var backpressure = true
+	// var err error
+	// if c.useClientAffinity && c.hnator != nil && c.partitionReplicas != nil && c.procedureInfos != nil {
+	// 	nc, backpressure, err =
+	// 		c.getConnByCA(c.connected, c.hnator, &c.partitionMasters, c.partitionReplicas, c.procedureInfos, pi)
+	// }
+	// if err != nil && !backpressure && nc != nil {
+	// 	// nc.submit(pi)
+	// } else {
+	// 	// c.allNcsPiCh <- pi
+	// }
+	return nc.submit(pi)
```

Comment:
Is the current plan for this pull request not using client affinity?

Reply:
Client affinity will be revisited in another PR. This was just supposed to resolve the issue issue (discussed elsewhere)

So, I have move a lot of code around and commented chunks that I believe will be relevant when I move to client affinity.